docs/versioned/admin/editing-configmaps.md

```diff
@@ -67,40 +66,18 @@ Define ConfigMaps in YAML or JSON files and store them in a version control syst
         db.port=5432
     ```
 
-- Commit these files to a Git repository to track changes over time.
+### Git recommendations
 
-### Versioning ConfigMaps in Git
+In addition to diligent usage of commit messages, here are some suggestions for ConfigMaps in GitHub:
 
-- Use meaningful commit messages to describe changes to ConfigMap data (e.g., "Updated app.properties to version 1.2.4").
+- Centralize ConfigMaps: Store all ConfigMaps in a dedicated directory in your Git repository (e.g., `k8s/configmaps/`).
 - Tag commits in Git with version numbers (e.g., `git tag config-v1.2.3`) to mark specific ConfigMap versions.
-- Use branches for different environments (e.g., `dev`, `staging`, `prod`) or feature-specific ConfigMap changes.
-
-### Track Object Version in ConfigMap
-
-- Include a version field in the ConfigMap’s data to explicitly track the version of the application or configuration it represents.
-
-    ```yaml
-    data:
-    app.properties: |
-      version=1.2.3
-      # other settings
-    ```
-
-- Alternatively, use annotations in the ConfigMap’s metadata:
-
-    ```yaml
-    metadata:
-      name: my-app-config
-      annotations:
-        app-version: "1.2.3"
-    ```
-
-### Use GitOps for Deployment
-
 - Implement a GitOps workflow with tools like ArgoCD or Flux to synchronize ConfigMaps from Git to your Kubernetes cluster.
 - These tools detect changes in the Git repository and automatically apply them to the cluster, ensuring the deployed ConfigMap matches the versioned configuration in Git.
 
-### Immutable ConfigMaps with Versioned Names
+### Immutable ConfigMaps
+
+If you have services processing information that must not change, such as for finances and payment processing, consider using immutable ConfigMaps. This approach ensures ConfigMaps are not modified in-place, preserving historical versions.
```

Comment:
Immutable ConfigMaps don't work for the Knative configuration, because the Knative controllers look up specific ConfigMap names, and immutable ConfigMaps need a new name each time.